lots of red CIs looking like

```
FAILED tests/test_pg_specific.py::test_pg_settings_dialog - AssertionError: 
Not equal to tolerance rtol=1e-07, atol=0

Mismatched elements: 1 / 1 (100%)
Max absolute difference among violations: 0.12
Max relative difference among violations: 0.00059113
 ACTUAL: array(203.12)
 DESIRED: array(203.)
```

--- a/src/mne_qt_browser/_dialogs.py
+++ b/src/mne_qt_browser/_dialogs.py
@@ -395,44 +395,22 @@ class SettingsDialog(_BaseDialog):
         screen = _screen(self)
         px_height = screen.size().height()
         px_width = screen.size().width()
-        if dim == "height":
-            new_ht_val = self.mon_height_spinbox.value()
-
-            # Get new dpi
-            mon_units = self.current_monitor_units
-            mon_height_inch = _convert_physical_units(
-                new_ht_val, from_unit=mon_units, to_unit="inch"
+        if dim in ("height", "width"):
+            # Pixels are square, so either dimension on its own pins down the DPI, and
+            # the other one then follows from it
+            if dim == "height":
+                px, value = px_height, self.mon_height_spinbox.value()
+            else:
+                px, value = px_width, self.mon_width_spinbox.value()
+            self.mne.dpi = px / _convert_physical_units(
+                value, from_unit=self.current_monitor_units, to_unit="inch"
             )
-            dpi = px_height / mon_height_inch
-
-            # Find new width of monitor
-            with QSignalBlocker(self.mon_width_spinbox):
-                mon_width = self.mne.aspect_ratio * new_ht_val
-                self.mon_width_spinbox.setValue(mon_width)
-
-            self.mne.dpi = dpi
-            self.dpi_spinbox.setValue(self.mne.dpi)
-
+            self._show_monitor_size(px_height, px_width)
             self._update_spinbox_values(ch_type="all", source="unit_change")
 
-        elif dim == "width":
-            new_wd_value = self.mon_width_spinbox.value()
-
-            # Get new dpi
-            mon_units = self.current_monitor_units
-            mon_width_inch = _convert_physical_units(
-                new_wd_value, from_unit=mon_units, to_unit="inch"
-            )
-            dpi = px_width / mon_width_inch
-
-            # Find new height of monitor
-            with QSignalBlocker(self.mon_height_spinbox):
-                mon_height = new_wd_value / self.mne.aspect_ratio
-                self.mon_height_spinbox.setValue(mon_height)
-
-            self.mne.dpi = dpi
-            self.dpi_spinbox.setValue(self.mne.dpi)
-
+        elif dim == "dpi":
+            self.mne.dpi = self.dpi_spinbox.value()
+            self._show_monitor_size(px_height, px_width)
             self._update_spinbox_values(ch_type="all", source="unit_change")
 
         elif dim == "unit_change":
@@ -455,56 +433,38 @@ class SettingsDialog(_BaseDialog):
 
             self.current_monitor_units = new_units
 
-        elif dim == "dpi":
-            new_value = self.dpi_spinbox.value()
-            self.mne.dpi = new_value
-            mon_units = self.current_monitor_units
-
-            with QSignalBlocker(self.mon_height_spinbox):
-                # px_height is already in the device-independent pixels that
-                # physicalDotsPerInch (and hence mne.dpi) counts, so no devicePixelRatio
-                mon_height_inch = px_height / new_value
-                self.mon_height_spinbox.setValue(
-                    _convert_physical_units(
-                        mon_height_inch, from_unit="inch", to_unit=mon_units
-                    )
-                )
-
-            with QSignalBlocker(self.mon_width_spinbox):
-                mon_width_inch = px_width / new_value
-                self.mon_width_spinbox.setValue(
-                    _convert_physical_units(
-                        mon_width_inch, from_unit="inch", to_unit=mon_units
-                    )
-                )
-
-            self._update_spinbox_values(ch_type="all", source="unit_change")
-
         else:
             raise ValueError(f"Unknown dimension: {dim}")
 
+    def _show_monitor_size(self, px_height, px_width):
+        """Show the monitor size implied by self.mne.dpi, and the DPI itself."""
+        for spinbox, px in (
+            (self.mon_height_spinbox, px_height),
+            (self.mon_width_spinbox, px_width),
+        ):
+            with QSignalBlocker(spinbox):
+                # px is already in the device-independent pixels that
+                # physicalDotsPerInch counts, so no devicePixelRatio here
+                spinbox.setValue(
+                    _convert_physical_units(
+                        px / self.mne.dpi,
+                        from_unit="inch",
+                        to_unit=self.current_monitor_units,
+                    )
+                )
+        with QSignalBlocker(self.dpi_spinbox):
+            self.dpi_spinbox.setValue(self.mne.dpi)
+
     def _reset_monitor_spinboxes(self):
         """Reset monitor spinboxes to expected values."""
-        mon_units = self.mon_units_cmbx.currentText()
-
-        # Get the screen size
         screen = _screen(self)
-        height_mm = screen.physicalSize().height()
-        width_mm = screen.physicalSize().width()
-
-        height_mon_units = _convert_physical_units(
-            height_mm, from_unit="mm", to_unit=mon_units
-        )
-        width_mon_units = _convert_physical_units(
-            width_mm, from_unit="mm", to_unit=mon_units
-        )
-
+        # physicalDotsPerInch averages the X and Y densities, which are two estimates
+        # of the same thing since pixels are square. Derive the sizes shown from that
+        # average rather than from physicalSize, which the two disagree about whenever
+        # it is rounded or, on a virtual display, invented -- otherwise re-entering a
+        # shown size would silently change the DPI.
         self.mne.dpi = screen.physicalDotsPerInch()
-
-        # Set the spinbox values as such
-        self.mon_height_spinbox.setValue(height_mon_units)
-        self.mon_width_spinbox.setValue(width_mon_units)
-        self.dpi_spinbox.setValue(self.mne.dpi)
+        self._show_monitor_size(screen.size().height(), screen.size().width())
 
         # Update sensitivity spinboxes
         self._update_spinbox_values(ch_type="all", source="unit_change")

--- a/src/mne_qt_browser/_pg_figure.py
+++ b/src/mne_qt_browser/_pg_figure.py
@@ -328,12 +328,9 @@ class MNEQtBrowser(BrowserBase, QMainWindow, metaclass=_PGMetaClass):  # type: i
         self.mne.scale_factor = 1
         # Factor of the scale factor that is due to butterfly mode
         self.mne.butterfly_scale = 1.0
-        # DPI
+        # DPI (the X/Y average, see SettingsDialog._reset_monitor_spinboxes)
         screen = QApplication.primaryScreen()
         self.mne.dpi = screen.physicalDotsPerInch()
-
-        # Aspect ratio
-        self.mne.aspect_ratio = screen.geometry().width() / screen.geometry().height()
         # Stores channel types for butterfly mode
         self.mne.butterfly_type_order = [
             tp for tp in DATA_CH_TYPES_ORDER if tp in self.mne.ch_types

--- a/src/mne_qt_browser/_utils.py
+++ b/src/mne_qt_browser/_utils.py
@@ -26,6 +26,7 @@ qsettings_params = {
 
 _unit_per_inch = dict(mm=25.4, cm=2.54, inch=1.0)
 
+
 # Butterfly mode draws traces at half amplitude (like the matplotlib backend, gh-276),
 # which equally shortens the scalebar and the value it stands for
 BUTTERFLY_SCALE = 0.5
@@ -140,7 +141,6 @@ def _calc_data_unit_to_physical(widget, units="mm"):
         return 0
 
     # Get the screen DPI
-    # px_per_in = QApplication.primaryScreen().logicalDotsPerInch()
     px_per_in = widget.mne.dpi
 
     # Convert to inches

--- a/tests/test_pg_specific.py
+++ b/tests/test_pg_specific.py
@@ -489,17 +489,30 @@ def test_pg_settings_dialog(raw_orig, pg_backend):
     orig_sens = ch_sens_spinbox.value()
     fig.mne.fig_settings.mon_height_spinbox.setValue(orig_mon_height / 2)
     QTest.keyPress(fig.mne.fig_settings.mon_height_spinbox.lineEdit(), Qt.Key_Return)
-    fig.mne.fig_settings.mon_width_spinbox.setValue(orig_mon_width / 2)
-    QTest.keyPress(fig.mne.fig_settings.mon_width_spinbox.lineEdit(), Qt.Key_Return)
+    assert_allclose(
+        fig.mne.fig_settings.dpi_spinbox.value(), orig_mon_dpi * 2, rtol=1e-3
+    )
+    assert_allclose(
+        fig.mne.fig_settings.mon_width_spinbox.value(), orig_mon_width / 2, rtol=1e-3
+    )
     assert ch_sens_spinbox.value() != orig_sens
 
-    # Re-entering the shown DPI is a no-op, i.e. it is the inverse of entering a
-    # monitor size
-    fig.mne.fig_settings._reset_monitor_spinboxes()
-    QTest.keyPress(fig.mne.fig_settings.dpi_spinbox.lineEdit(), Qt.Key_Return)
-    assert_allclose(fig.mne.fig_settings.mon_height_spinbox.value(), orig_mon_height)
-    assert_allclose(fig.mne.fig_settings.mon_width_spinbox.value(), orig_mon_width)
-    assert_allclose(fig.mne.fig_settings.dpi_spinbox.value(), orig_mon_dpi)
+    # Reset leaves height, width and DPI mutually consistent -- pixels are square, even
+    # where the physical size Qt reports is not -- so re-entering any one of them is a
+    # no-op. rtol covers the spinboxes rounding to two decimals.
+    for box in ("mon_height_spinbox", "mon_width_spinbox", "dpi_spinbox"):
+        fig.mne.fig_settings._reset_monitor_spinboxes()
+        QTest.keyPress(getattr(fig.mne.fig_settings, box).lineEdit(), Qt.Key_Return)
+        assert_allclose(
+            fig.mne.fig_settings.mon_height_spinbox.value(), orig_mon_height, rtol=1e-3
+        )
+        assert_allclose(
+            fig.mne.fig_settings.mon_width_spinbox.value(), orig_mon_width, rtol=1e-3
+        )
+        assert_allclose(
+            fig.mne.fig_settings.dpi_spinbox.value(), orig_mon_dpi, rtol=1e-3
+        )
+        assert_allclose(ch_sens_spinbox.value(), orig_sens, rtol=1e-3)
 
     # Monitor settings reset button works
     fig.mne.fig_settings._reset_monitor_spinboxes()

